commit 0db7a007594bf11031ea133d0df4ec7d3d88e82c
Author: Alexander V. Chernikov <melifaro@ipfw.ru>
Date:   Fri Mar 1 16:19:47 2013 +0000

    Permit loopback interfaces to be used as router id source

diff --git a/nest/iface.c b/nest/iface.c
index da79b21..a12cbdc 100644
--- nest/iface.c
+++ nest/iface.c
@@ -589,7 +589,7 @@ if_choose_router_id(struct iface_patt *mask, u32 old_id)
   WALK_LIST(i, iface_list)
     {
       if (!(i->flags & IF_ADMIN_UP) ||
-	  (i->flags & (IF_IGNORE | IF_SHUTDOWN)))
+	  (i->flags & IF_SHUTDOWN))
 	continue;
 
       WALK_LIST(a, i->addrs)
@@ -612,6 +612,9 @@ if_choose_router_id(struct iface_patt *mask, u32 old_id)
 	  if (mask && !iface_patt_match(mask, i, a))
 	    continue;
 
+	  if ((i->flags & IF_IGNORE) && !mask)
+	    continue;
+
 	  /* No pattern or pattern matched */
 	  if (!b || ipa_to_u32(a->ip) < ipa_to_u32(b->ip))
 	    b = a;
